### ISSUE TITLE
Insufficient data passed when error occurs

### ISSUE THREAD

**Issue by pgsandstrom** [marcin-piela/react-fetching-library#24]
Im running code similar to this:

```
    doQuery({
      method: 'POST',
      endpoint: '/v1/bil/drivmedelspris/prislinjen',
      body: prislinjenRapport
    }).then((value) => {
      console.log(value)
    })
```

This is a CORS request, so I can see that a OPTION and POST takes place, both returning 204 no content. The response that is logged is of this type:

`{"error":true,"errorObject":{ SyntaxError: Unexpected end of JSON input at... }}`

As I see it, there is two problems here. Both that the error occurs, and that I receive insufficient information in the error that is thrown. I cant see the status code of the request or anything like that.

**Comment by marcin-piela** [comment]
@pgsandstrom thanks for that issue. I see a problem in `Client` implementation and thats why error (Unexpected end of JSON) occurred. It's connected with empty body response, cause client expected always a JSON here. I'll fix it today.

**Comment by pgsandstrom** [comment]
Thanks for the quick reply! Awesome! I was also digging through the code and saw this :) The code seems to assume that both request and response body is json (by adding 'application/json; charset=utf-8' as Content-Type to request). Is there a reason for this that I am missing? While I really wished that I only had to work in json, this is not my current reality.

**Comment by marcin-piela** [comment]
`Content-Type: application/json; charset=utf-8` - it's the default header in a library when other is not provided.

It can be overwritten in action definition:
```
{
 method: 'GET',
 endpoint: '/test',
 headers: {
    "Content-type": "--here--",
 }
}
```

or in request interceptor:

```
export const requestHeaderInterceptor =( ) => async action => {
  return {
    ...action,
     headers: {
        "Content-type": "--here--",
      }
  };
};
```

**Comment by marcin-piela** [comment]
Released in 1.2.1 version
